boot/bootutil/include/bootutil/image.h

```diff
@@ -142,7 +142,7 @@ struct image_tlv {
     uint16_t it_len;    /* Data length (not including TLV header). */
 };
 
-#define IS_ENCRYPTED(hdr) ((hdr)->ih_flags & IMAGE_F_ENCRYPTED)
+#define IS_ENCRYPTED(hdr) (((hdr)->ih_flags & (IMAGE_F_ENCRYPTED)) && (hdr->ih_magic == IMAGE_MAGIC))
```

Comment:
I would suggest trying something like:

```
#define IS_ENCRYPTED(hdr) (((hdr)->ih_flags & IMAGE_F_ENCRYPTED) && ((hdr)->ih_flags != 0xffffffff))
```

Reply:
This kind of implies that we won't ever use all of the flag bits. That might be a reasonable thing to do, especially if we decide to make a bit reserved, and indicate it must always be 0.

Additionally, not all flash devices use 0xffffffff as the erased value. Is there a particular reason to not check the magic? Arguably, that should be done first by the code using this, rather than every time the flags are checked, though.

Reply:
A possible fix is to clear ih_flags after having read it from flash . A  check of  hdr magic is added in function boot_read_image_header from swap_move.c or swap_scratch.c, if magic is not present all ih_flags are cleared.

Reply:
Updated patch fixes installation of encrypted file in secondary slot to an empty primary slot in SWAP_USING_MOVE  and SWAP_USING_SCRATCH